src/types/index.d.ts

```diff
@@ -1,16 +1,24 @@
 /* eslint-disable camelcase */
 import type { GridLayerOptions, Coords, DoneCallback } from "leaflet";
 
-export type PixelValueToColorFn = (values: number[]) => string;
+export type PixelValuesToColorFn = (values: number[]) => string;
 
-export interface GeorasterLayerOptions extends GridLayerOptions {
-  georasters: Georaster[];
-  georaster: Georaster;
+interface GeoRasterLayerOptions_CommonOptions extends GridLayerOptions {
   resolution?: number;
-  debugLevel?: 0 | 1;
-  pixelValuesToColorFn?: PixelValueToColorFn;
+  debugLevel?: 0 | 1 | 2;
+  pixelValuesToColorFn?: PixelValuesToColorFn;
 }
 
+// Ensures at least one of the georaster[s] options is defined while being ok the other is not
```

Comment:
Wow this is awesome.

Reply:
Super cool right. I still need to look at it a few times to make sure I understand what it's doing. But its very powerful I find.